Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidTodoCommentsAnalyzer.cs

```diff
@@ -38,12 +39,41 @@ public override void Analyze()
 			foreach (SyntaxTrivia comment in comments)
 			{
 				var commentText = comment.ToString();
-				if (commentText.ToUpperInvariant().Contains("TODO"))
+				if (ContainsTodoAsWord(commentText))
 				{
 					Location location = comment.GetLocation();
 					ReportDiagnostic(location);
 				}
 			}
 		}
+
+		private static bool ContainsTodoAsWord(string text)
+		{
```

Comment:
This is a lot of code for what we're trying to accomplish. Would something like this be better?

private static readonly Regex TodoWordRegex =
    new(@"(?<!\w)TODO(?!\w)", RegexOptions.IgnoreCase | RegexOptions.CultureInvariant | RegexOptions.Compiled);

private static bool ContainsTodoAsWord(string text)
    => TodoWordRegex.IsMatch(text);

(I can explore sourcegenerator later. for now, maybe do the slow regex after an otherwise unnecessary Contains("todo") check?)